src/actions/notifications/index.ts

```diff
@@ -1,41 +1,136 @@
+import { ThunkAction } from 'src/store'
+import { Notification } from 'src/lib/notifications'
 import {
-  CLEAR_NOTIFICATIONS,
-  REMOVE_NOTIFICATION,
+  SET_ACTIVE_NOTIFICATION,
   SCHEDULE_NOTIFICATION,
-} from '../../reducers/notifications'
-import { randomBytes } from 'crypto'
-import {
-  Notification,
-  NotificationMessage,
-  NotificationScope,
-  NotificationSeverity,
-  NotificationType,
-} from '../../reducers/notifications/types'
-
-export const removeNotification = (notification: Notification) => ({
-  type: REMOVE_NOTIFICATION,
-  value: notification,
-})
+  REMOVE_NOTIFICATION,
+  CLEAR_NOTIFICATIONS,
+} from 'src/reducers/notifications'
 
-export const scheduleNotification = (notification: Notification) => ({
-  type: SCHEDULE_NOTIFICATION,
-  value: notification,
-})
 
-export const clearAllNotifications = () => ({
-  type: CLEAR_NOTIFICATIONS,
+export const setActiveNotification = (
+  notification: Notification | null,
+  expiry?: number,
+) => ({
+  type: SET_ACTIVE_NOTIFICATION,
+  notification,
+  expiry,
 })
 
-export const infoNotification = (
-  message: NotificationMessage,
-): Notification => ({
-  uid: randomBytes(4).toString('hex'), // TODO abstract
-  type: NotificationType.info,
-  message,
-  severity: NotificationSeverity.medium,
-  scope: NotificationScope.global,
-  dismissible: true,
-  autoDismissMs: 3000,
-  handleConfirm: removeNotification,
-  handleDismiss: removeNotification,
-})
+export const scheduleNotification = (
+  notification: Notification,
+): ThunkAction => dispatch => {
+  dispatch({
+    type: SCHEDULE_NOTIFICATION,
+    notification,
+  })
+  return dispatch(updateNotificationsState)
+}
+
+export const removeNotification = (
+  notification: Notification,
+): ThunkAction => dispatch => {
+  dispatch({
+    type: REMOVE_NOTIFICATION,
+    notification,
+  })
+  return dispatch(updateNotificationsState)
+}
+
+export const invokeInteract = (
+  notif: Notification,
+): ThunkAction => async dispatch => {
+  let keepNotification = false
+  const { interact } = notif
+  if (interact && interact.onInteract) {
+    keepNotification = (await interact.onInteract()) === true
+  }
+  if (!keepNotification) return dispatch(removeNotification(notif))
+}
+
+export const invokeDismiss = (
+  notif: Notification,
+): ThunkAction => async dispatch => {
+  const { dismiss } = notif
+  if (dismiss && typeof dismiss === 'object' && dismiss.onDismiss)
```

Comment:
Why is the extra check if dismiss is an object needed?

Reply:
Even if it's a boolean, i.e. false, `false.onDismiss` is undefined, which would be ok in this case.

Reply:
Ahh, nvm, I see this is to keep TS happy :(

Reply:
yes to keep typescript happy, but the first check (for just `dismiss`) can be eliminated :D